pizza_cutter_metadetect/masks.py

```diff
@@ -0,0 +1,450 @@
+        A directory with keys 'min_row', 'max_row', 'min_col', and 'max_col'.
+    """
+    slice_dim = central_size + 2*buffer_size
+    if orig_start_row == 0:
+        min_slice_row = 0
+        max_slice_row = slice_dim - buffer_size
+    elif orig_start_row + slice_dim == coadd_dims[0]:
+        min_slice_row = buffer_size
+        max_slice_row = slice_dim
+    else:
+        min_slice_row = buffer_size
+        max_slice_row = slice_dim - buffer_size
+
+    if orig_start_col == 0:
+        min_slice_col = 0
+        max_slice_col = slice_dim - buffer_size
+    elif orig_start_col + slice_dim == coadd_dims[1]:
+        min_slice_col = buffer_size
+        max_slice_col = slice_dim
+    else:
+        min_slice_col = buffer_size
+        max_slice_col = slice_dim - buffer_size
+
+    return {
+        "min_row": min_slice_row,
+        "max_row": max_slice_row,
+        "min_col": min_slice_col,
+        "max_col": max_slice_col,
+    }
+
+
+def _build_base_tile_mask(*, wcs, position_offset, coadd_dims, healpix_nside, info):
+    dpix = 250
+
+    geom_prims = []
+
+    n_x = coadd_dims[1] // dpix
+    if n_x * dpix < coadd_dims[1]:
+        n_x += 1
+
+    n_y = coadd_dims[0] // dpix
+    if n_y * dpix < coadd_dims[0]:
+        n_y += 1
+
+    for xind in tqdm.trange(n_x, ncols=79, desc="building base tile mask"):
+        xstart = xind * dpix
+        xend = xstart + dpix
+        if xend >= coadd_dims[1]:
+            xend = coadd_dims[1]-1
+        xlocs = np.array([
+            xstart, xend, xend, xstart
+        ])
+
+        for yind in range(n_y):
+            ystart = yind * dpix
+            yend = ystart + dpix
+            if yend >= coadd_dims[0]:
+                yend = coadd_dims[0]-1
+            ylocs = np.array([
+                ystart, ystart, yend, yend,
+            ])
+
+            ra, dec = wcs.image2sky(xlocs + position_offset, ylocs + position_offset)
+
+            # skip polygons outside the unique tile bounds
+            if not np.any(in_unique_coadd_tile_region(
+                ra=ra,
+                dec=dec,
+                crossra0=info['crossra0'],
+                udecmin=info['udecmin'],
+                udecmax=info['udecmax'],
+                uramin=info['uramin'],
+                uramax=info['uramax'],
+            )):
+                continue
+
+            ra, dec = truncate_to_unique_coadd_tile_region(
+                ra=ra,
+                dec=dec,
+                crossra0=info['crossra0'],
+                udecmin=info['udecmin'],
+                udecmax=info['udecmax'],
+                uramin=info['uramin'],
+                uramax=info['uramax'],
+            )
+            geom_prims.append(
+                healsparse.Polygon(
+                    ra=ra,
+                    dec=dec,
+                    value=MASK_INTILE,
+                )
+            )
+
+    hs_msk_prims = healsparse.HealSparseMap.make_empty(
+        32, healpix_nside, np.int32, sentinel=0
```

Comment:
nside=32 for the coverage mask is a good compromise for an all-sky mask.  However, if you're building one slice at a time at very high resolution this is not very efficient.  You should choose a coverage nside that gives an area close to or a bit larger than the typical area of each tile.  However, you shouldn't go higher than 128 (and healsparse will give you a warning).  My guess is that you want 128 for these maps.

Reply:
What implications would 128 have for merging the masked files later at lower resolution?

Reply:
It's not hard to convert the coverage nside.  And https://github.com/LSSTDESC/healsparse/blob/master/healsparse/cat_healsparse_files.py will combine files and normalize the coverage nside.  If we want to degrade them at the same time we can add that.

Reply:
fixed